lib/model.js

```diff
@@ -4214,10 +4214,14 @@ Model.aggregate = function aggregate(pipeline, options, callback) {
 
   const aggregate = new Aggregate(pipeline || []);
   aggregate.model(this);
-
   if (options != null) {
     aggregate.option(options);
   }
+  const defaultAllowDiskUse = this.base.options.allowDiskUse != null ? this.base.options.allowDiskUse : false;
```

Comment:
For consistency, I'd prefer if we implemented this the same way we implemented global `maxTimeMS` using the [`applyGlobalMaxTimeMS()` helper](https://github.com/Automattic/mongoose/blob/80245edf5aeab304411681035f7f4153dbd2b692/lib/helpers/query/applyGlobalMaxTimeMS.js). Two reasons: 1) consistency in not applying default options until right before execution, and 2) `allowDiskUse` is also an option for queries, so we need to make global `applyDiskUse` apply to both aggregations and queries.

Reply:
Done, I initially wanted to use the option right before execution, but that made testing a little bit more challenging.
I used Sinon to add a spy on the native driver `aggregate `method to get the options that end up there. It will help with some other tests as well.